src/components/Location/LocationTree.tsx

```diff
@@ -11,16 +11,57 @@ interface LocationPathProps {
   showTimeline?: boolean;
 }
 
-function getLocationHierarchy(location: LocationList): LocationList[] {
-  const hierarchy: LocationList[] = [];
-  let current: LocationList | undefined = location;
-
-  while (current?.id) {
-    hierarchy.unshift(current);
-    current = current.parent;
-  }
+interface LocationNodeProps {
+  location: LocationList;
+  depth: number;
+  isLast: boolean;
+  datetime?: string;
+}
 
-  return hierarchy;
+function LocationNode({
+  location,
+  depth,
+  isLast,
+  datetime,
+}: LocationNodeProps) {
+  return (
+    <>
+      {location.parent?.id && (
+        <LocationNode
+          location={location.parent}
+          depth={depth - 1}
+          isLast={false}
+          datetime={datetime}
+        />
+      )}
```

Comment:
let's keep the node inside the parent node so that we can skip passing depth and calculations for padding